### ISSUE TITLE
Reading the dataset without tensorflow

### ISSUE THREAD

**Issue by l3robot** [deepmind/gqn-datasets#5]
Hi,

I have searched for quite a long time now and I'm looking for a fast and efficient way of reading your dataset without tensorflow. I could indeed use a minimum of tensorflow code, but what I've seen is that we are forced to run the `DataReader.read` method inside a tensorflow session. 

I've looked into solutions like using this code [https://github.com/pgmmpk/tfrecord](https://github.com/pgmmpk/tfrecord), but it's handled a different way and the data wrongly decoded. 

Do you have recommendations on how to use the dataset without or with minimal tensorflow code?

Thanks in advance.

**Comment by l3robot** [comment]
I managed to convert it and I have a script if you are interested.

**Comment by fabioviola** [comment]
Sorry for the slow reply - great to hear you found a solution.
Why don't you add a link to the solution in this thread? That might be helpful for other users.
Thanks

**Comment by l3robot** [comment]
No worries, yeah sure, here is the link https://github.com/l3robot/gqn_datasets_translator

**Comment by l3robot** [comment]
Hi again! 
I used my script for many days now, but just remarked something. Where you are mentioning 2M scenes in the DeepMind paper for rooms, there are actually >8M scenes in the released dataset. How I understand it is that each record contains multiple scenes. Maybe I'm wrong about the inner structure of the tf-records. May you clarify the dataset structure of each record? 
Thanks a lot.

**Comment by l3robot** [comment]
Hi again!
I used my script for many days now, but just remarked something with the rooms datasets. Where you are mentioning 2M scenes in the DeepMind paper for rooms, there are actually >8M scenes in the released dataset. How I understand it is that each record contains multiple scenes. Maybe I'm wrong about the inner structure of the tf-records. May you clarify the dataset structure of each record?
Thanks a lot.

**Comment by fabioviola** [comment]
Hello Emile,

The room datasets we open sourced comprise multiple tf records, ~2k in train set, ~200 in test set.
Each record contains data from 5K separate scenes, i.e. frames and camera coordinates; each time a scene is used a random subset of its views is selected to form a query.
During training the number of views used to compose query is randomized.

Tip: you can count how many entries are in a given tf record using this snippet:

```
count_entries = lambda filename: sum(1 for _ in tf.python_io.tf_record_iterator(filename))
```

Thank you for asking clarification about the 2M reference in the appendix, I assume you refer to:

```
We sample 2 million scenes and 5 images per scene at a resolution of 64x64 in order to construct the dataset.
```

The 2M number refers to the number of steps used in training: we generate 2 million scene _batches_ worth of data on the fly by sampling with replacement from the pre-rendered scenes stored in the tf-records.

Please feel free to reopen the issue if you have further questions.

**Comment by l3robot** [comment]
Oh! thanks a lot for this more than complete answer. This will help a lot! Yes that was exactly what I was referring to, makes lot of sense.

**Comment by jenspetersen** [comment]
@l3robot nice work with the converter, all PyTorch implementations I saw use it :) Still, I can imagine it's pretty annoying if everyone needs to convert the data before they can use it with anything other than Tensorflow. So my question to the DeepMind guys is: Would you be willing to host numpy versions of the datasets? I could provide those, but I'm having a hard time finding a good place to host the data...
